### ISSUE TITLE
Vala to CS example

### ISSUE THREAD

**Issue by gavr123456789** [gircore/gir.core#33]
Were there any experiments with creating an object in Vala and importing it to C#?  
Or just forecasts of how well it will work.  
  
(Also interested in Meson)

**Comment by badcel** [comment]
No I did not test vala. The generator expects a gir file to generate the wrapper. The vala compiler can output a gir file so I think it should work. There could be small issues with the types, if there are some bits which are missing in the type resolver.

I want to make this project like an entrypoint for C# developers to use the GObject-Stack. Do you have a special vala library in mind which is missing in this repository?  Otherwise I think it would be great to have an example for a custom vala library, so that everybody can write a wrapper for their own library, allowing to stay in the same C# GObject type hierarchy like the rest of the projects hosted here.

If there is a need there could even be a special dotnet tool to allow the generation of bindings from arbitrary gir files. 
If the bindings generated from a vala gir file is failing for you I would consider this a bug in this project.

**Comment by gavr123456789** [comment]
> If there is a need there could even be a special dotnet tool to allow the generation of bindings from arbitrary gir files.
  
Do you already have any plans for this? That would be so cool.  
  
> Do you have a special vala library in mind which is missing in this repository?   
  
No, but I just know a few GTK libraries(for example [1](https://notabug.org/grindhold/libgtkflow), [2](https://github.com/lcallarec/live-chart) [3](https://gitlab.com/doublehourglass/dnd_order_list_box)) written in Vala. You can also use Vala to write application bottlenecks(since it generates C) and there still no better language to write something for GTK.

**Comment by badcel** [comment]
> Do you already have any plans for this? That would be so cool.

No, it should not be very hard. There are a lot of projects which provide their own tools. But  with the current implementation it would only generate the lower layer. I would like any contribution in this direction.

> No, but I just know a few GTK libraries(for example 1, 2 3)

I didn't know these, they are very interesting!

Next I wanted to check if I could get s.th. to work with gtk4.

**Comment by badcel** [comment]
There should be some kind of `dotnet tool` support to allow 3rd party developers to create new GObject C# bindings based on top of our stack. I changed the title to make this a little bit clearer.

**Comment by badcel** [comment]
Another library is https://github.com/elementary/granite from elementary os

**Comment by badcel** [comment]
#1078 is a little more concrete so let's close this.
